Casks/g/graalvm-jdk.rb

```diff
@@ -13,10 +13,10 @@
 
   livecheck do
     url "https://www.oracle.com/java/technologies/downloads/"
-    regex(/graalvm\s+for\s+jdk\s+(\d+(?:\.\d+)?)/i)
+    regex(/graalvm\s+for\s+jdk\s+(\d+(?:\.\d+)+)/i)
```

Comment:
```suggestion
    regex(/GraalVM\s+for\s+JDK\s+v?(\d+(?:\.\d+)*)\s+downloads/im)
```

Looking back at previous snapshots of the page, the heading was simply "GraalVM for JDK 21 downloads" for the first 21 release, so I agree that using `(\d+(?:\.\d+)*)` is appropriate here.

The full version string only appears to be found in the "GraalVM for JDK 21.0.1 downloads" heading on the page, so there's something to be said for specifically matching that. With the existing regex, livecheck will return both the major version and the full version for this release: 21, 17, 21.0.1, 17.0.9. With the suggestion above, livecheck will only return the full versions: 21.0.1, 17.0.9. [Of course, when 22 is released, livecheck will return 22 as the full version.]

Besides that, the suggestion above adds the multiline (`m`) flag (since we're matching loose text that may end up split across multiple lines in the HTML file) and tidies up the regex a bit (it's a case-insensitive regex but using more than lowercase helps readers to understand what the text we're aiming to match looks like on the page).